src/WebJobs.Extensions.DurableTask.Scale/AzureStorage/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -0,0 +1,270 @@
+        private readonly ILoggerFactory loggerFactory;
+        private readonly bool useSeparateQueueForEntityWorkItems;
+        private readonly bool inConsumption; // If true, optimize defaults for consumption
+        private AzureStorageDurabilityProvider defaultStorageProvider;
+
+        // Must wait to get settings until we have validated taskhub name.
+        private bool hasValidatedOptions;
+        private AzureStorageOrchestrationServiceSettings defaultSettings;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="clientProviderFactory"></param>
+        /// <param name="nameResolver"></param>
+        /// <param name="loggerFactory"></param>
+        /// <param name="platformInfo"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public AzureStorageDurabilityProviderFactory(
+            IOptions<DurableTaskOptions> options,
+            IStorageServiceClientProviderFactory clientProviderFactory,
+            INameResolver nameResolver,
+            ILoggerFactory loggerFactory,
+#pragma warning disable CS0612 // Type or member is obsolete
+            IPlatformInformation platformInfo)
+#pragma warning restore CS0612 // Type or member is obsolete
+        {
+            // this constructor may be called by dependency injection even if the AzureStorage provider is not selected
+            // in that case, return immediately, since this provider is not actually used, but can still throw validation errors
+            if (options.Value.StorageProvider.TryGetValue("type", out object value)
+                && value is string s
+                && !string.Equals(s, this.Name, StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            this.options = options.Value;
+            this.clientProviderFactory = clientProviderFactory ?? throw new ArgumentNullException(nameof(clientProviderFactory));
+            this.nameResolver = nameResolver ?? throw new ArgumentNullException(nameof(nameResolver));
+            this.loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+
+            this.azureStorageOptions = new AzureStorageOptions();
+            this.inConsumption = platformInfo.IsInConsumptionPlan();
+
+            // The consumption plan has different performance characteristics so we provide
+            // different defaults for key configuration values.
+            int maxConcurrentOrchestratorsDefault = this.inConsumption ? 5 : 10 * Environment.ProcessorCount;
+            int maxConcurrentActivitiesDefault = this.inConsumption ? 10 : 10 * Environment.ProcessorCount;
+            int maxConcurrentEntitiesDefault = this.inConsumption ? 10 : 10 * Environment.ProcessorCount;
+            int maxEntityOperationBatchSizeDefault = this.inConsumption ? 50 : 5000;
+
+            if (this.inConsumption)
```

Comment:
I wonder if we should keep the logic from line 72 to line 92. Cause when I checked the TriggerMetadata json payload from SC I didn't find info about runtime. Also I think this basically is not related with scaling?